.github/workflows/deploy.yml

```
@@ -0,0 +1,218 @@
+name: Deploy Lambda Function
+
+on:
+  push:
+    branches: [ prod, stg ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  AWS_REGION: us-east-1
+  LAMBDA_FUNCTION_NAME_PROD: benefits-screening-api
+  LAMBDA_FUNCTION_NAME_STG: benefits-screening-api-stg
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v4
+      with:
+        version: "latest"
+
+    - name: Install dependencies
+      run: uv sync --dev
+
+    - name: Run tests
+      run: uv run pytest
+
+    - name: Run linting (optional)
+      run: |
+        if uv run which ruff >/dev/null 2>&1; then
+          uv run ruff check .
+        else
+          echo "Ruff not installed, skipping linting"
+        fi
+      continue-on-error: true
+
+  deploy-production:
+    needs: test
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v4
+      with:
+        version: "latest"
+
+    - name: Install production dependencies
+      run: uv sync --no-dev
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: ${{ env.AWS_REGION }}
+
+    - name: Create deployment package
+      run: |
+        # Create a clean directory for the package
+        mkdir -p package
+        
+        # Export dependencies to package directory
+        uv export --no-dev --format requirements-txt > requirements.txt
+        pip install -r requirements.txt -t package/
+        
+        # Copy your source code to package (adjust paths as needed for your project structure)
```

```suggestion
        # Copy your source code to package
```
Did you check that the paths are correct with uv?

---

Respond to comment above^